src/Git/Index/Update.hs

```diff
@@ -60,6 +69,41 @@ updateIndex relpath = do
       Normal ie -> Just ie
       _ -> Nothing
 
+parseContent :: MonadFail m => Parser a -> Path.AbsFile -> IO (m a)
```

Comment:
Did you stop this being quite such an abstract `Monad` because you wanted to wrap it in the `try` on line 80? Would it not be better to guard the exception inside this function and simply give you back whatever `MonadFail` you wanted it you can't open the file?

Reply:
I want to catch one of the error conditions in here (namely the file not existing) and handle it, but not the parse failure. So it ceased to be `MonadFail` and became `MonadError <Something>` so I bailed it on the basis that I didn't know what the `Something` should be (it isn't `GitError` because we expect to handle the error, it could be something like `Either GitError ()` but that seemed odd too.